### PR TITLE
[FLINK-15081][docs-zh]Translate "Concepts & Common API" page of Table…

### PR DESCRIPTION
**Pull request by CrazyTomatoOo**
## What is the purpose of the change

Translate "Concepts & Common API" page of Table.

## Brief change log

Translation work has been finished without code annotation.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`:  no
  - The serializers:  no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature?  no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Comment by wuchong** [comment]
Hi @libenchao , could you help to review this?

**Comment by wuchong** [comment]
@CrazyTomatoOo could you resolve the conflict if you have time?

File: docs/dev/table/common.zh.md [hunk 1]
```diff
@@ -723,27 +702,27 @@ table_env.sql_update(
...
+* 通过在 `TableEnvironment` 中注册[结果表](#register-a-table)并在 SQL 查询的 `FROM` 子句中引用它，可以在 Table API 查询的结果上定义 SQL 查询。
 
 {% top %}
 
-Emit a Table 
+发出表
 ------------
 
-A `Table` is emitted by writing it to a `TableSink`. A `TableSink` is a generic interface to support a wide variety of file formats (e.g. CSV, Apache Parquet, Apache Avro), storage systems (e.g., JDBC, Apache HBase, Apache Cassandra, Elasticsearch), or messaging systems (e.g., Apache Kafka, RabbitMQ). 
+`表`通过写入 `TableSink` 发出。`TableSink` 是一个通用接口，用于支持多种文件格式（如 CSV、Apache Parquet、Apache Avro）、存储系统（如 JDBC、Apache HBase、Apache Cassandra、Elasticsearch）或消息传递系统（如 Apache Kafka、RabbitMQ）。
```
Comment (libenchao) [inline docs/dev/table/common.zh.md]: 发出 -> 输出
消息传递 -> 消息队列

File: docs/dev/table/common.zh.md [hunk 2]
```diff
@@ -827,70 +806,70 @@ result.insert_into("CsvSinkTable")
 {% top %}
 
 
-Translate and Execute a Query
+解析与执行查询
```
Comment (libenchao) [inline docs/dev/table/common.zh.md]: how about “翻译” ？

File: docs/dev/table/common.zh.md [hunk 3]
```diff
@@ -827,70 +806,70 @@ result.insert_into("CsvSinkTable")
...
+* `表`被转换成`流数据`或者`批数据`时（参阅[与 DataStream 和 DataSet API 结合](#integration-with-datastream-and-dataset-api)）。
 
-Once translated, a Table API or SQL query is handled like a regular DataStream or DataSet program and is executed when `StreamExecutionEnvironment.execute()` or `ExecutionEnvironment.execute()` is called.
+解析完成后，Table API 或者 SQL 查询会被当做普通的流处理或批处理程序对待并且会在调用 `StreamExecutionEnvironment.execute()` 或 `ExecutionEnvironment.execute()` 的时候被执行。
 
 </div>
 
 <div data-lang="Blink planner" markdown="1">
-Table API and SQL queries are translated into [DataStream]({{ site.baseurl }}/dev/datastream_api.html) programs whether their input is streaming or batch. A query is internally represented as a logical query plan and is translated in two phases:
+Table API 和 SQL 查询会被转换成[流处理]({{ site.baseurl }}/zh/dev/datastream_api.html)程序不论它们的输入数据源是流式的还是批式的。查询在内部表示为逻辑查询计划，并被解析成两个阶段：
```
Comment (libenchao) [inline docs/dev/table/common.zh.md]: Table API 和 SQL 查询会被转换成[流处理]({{ site.baseurl }}/zh/dev/datastream_api.html)程序不论它们的输入数据源是流式的还是批式的
->
不论输入数据源是流式的还是批式的，Table API 和 SQL 查询都会被转换成[流处理]({{ site.baseurl }}/zh/dev/datastream_api.html)程序。

File: docs/dev/table/common.zh.md [hunk 4]
```diff
@@ -1091,17 +1070,17 @@ val dsTuple: DataSet[(String, Int)] = tableEnv.toDataSet[(String, Int)](table)
 
 {% top %}
 
-### Mapping of Data Types to Table Schema
+### 数据类型到 Table Schema 的映射
 
-Flink's DataStream and DataSet APIs support very diverse types. Composite types such as Tuples (built-in Scala and Flink Java tuples), POJOs, Scala case classes, and Flink's Row type allow for nested data structures with multiple fields that can be accessed in table expressions. Other types are treated as atomic types. In the following, we describe how the Table API converts these types into an internal row representation and show examples of converting a `DataStream` into a `Table`.
+Flink 的 DataStream 和 DataSet APIs 支持多样的数据类型。例如 Tuple（Scala 内置以及Flink Java tuple）、POJO 类型、Scala case class 类型以及 Flink 的 Row 类型等允许嵌套且有多个可在表的表达式中访问的字段的复合数据类型。其他类型被视为原子类型。 Composite types such as Tuples (built-in Scala and Flink Java tuples), POJOs, Scala case classes, and Flink's Row type allow for nested data structures with multiple fields that can be accessed in table expressions. 下面，我们讨论 Table API 如何将这些数据类型类型转换为内部 row 表示形式，并提供将流数据集转换成表的样例。
```
Comment (libenchao) [inline docs/dev/table/common.zh.md]: forgets to remove original english doc.

File: docs/dev/table/common.zh.md [hunk 5]
```diff
@@ -1091,17 +1070,17 @@ val dsTuple: DataSet[(String, Int)] = tableEnv.toDataSet[(String, Int)](table)
...
+数据类型到 table schema 的映射有两种方式：**基于字段位置**或**基于字段名称**。
 
-**Position-based Mapping**
+**基于位置映射**
 
-Position-based mapping can be used to give fields a more meaningful name while keeping the field order. This mapping is available for composite data types *with a defined field order* as well as atomic types. Composite data types such as tuples, rows, and case classes have such a field order. However, fields of a POJO must be mapped based on the field names (see next section). Fields can be projected out but can't be renamed using an alias `as`.
+基于位置的映射可在保持字段顺序的同时为字段提供更有意义的名称。这种映射方式可用于*具有确定的字段顺序*的复合数据类型以及原子类型。如 tuple、row 以及 case class 这些复合数据类型都有这样的字段顺序。然而，POJO 类型的字段则必须通过名称映射（参见下一章）。可以将字段投影出来，但不能使用`as`重命名。
 
-When defining a position-based mapping, the specified names must not exist in the input data type, otherwise the API will assume that the mapping should happen based on the field names. If no field names are specified, the default field names and field order of the composite type are used or `f0` for atomic types. 
+定义基于位置的映射时，输入数据类型中一定不能存在指定的名称，否则 API 会假定应该基于字段名称进行映射。如果未指定任何字段名称，则使用默认的字段名称和复合数据类型的字段顺序，或者使用 “ f0” 表示原子类型。
```
Comment (libenchao) [inline docs/dev/table/common.zh.md]: “ f0” -> `f0`

File: docs/dev/table/common.zh.md [hunk 6]
```diff
@@ -1191,9 +1170,9 @@ val table: Table = tableEnv.fromDataStream(stream, '_2 as 'myInt, '_1 as 'myLong
 </div>
 </div>
 
-#### Atomic Types
+#### 原子类型
 
-Flink treats primitives (`Integer`, `Double`, `String`) or generic types (types that cannot be analyzed and decomposed) as atomic types. A `DataStream` or `DataSet` of an atomic type is converted into a `Table` with a single attribute. The type of the attribute is inferred from the atomic type and the name of the attribute can be specified.
+Flink 将原始数据类型（`Integer`、`Double`、`String`）或者通用数据类型（不可再拆分的数据类型）视为原子类型。原子类型的`流数据集`或者`批数据集`会被转换成只有一条属性的`表`。属性的数据类型可以由原子类型推断出，还可以重新命名属性。
```
Comment (libenchao) [inline docs/dev/table/common.zh.md]: prefer “基础类型” to “原始类型”

File: docs/dev/table/common.zh.md [hunk 7]
```diff
@@ -1406,54 +1385,54 @@ val table: Table = tableEnv.fromDataStream(stream, 'name as 'myName)
...
 
-Query Optimization
+查询优化
 ------------------
 
 <div class="codetabs" markdown="1">
 <div data-lang="Old planner" markdown="1">
 
-Apache Flink leverages Apache Calcite to optimize and translate queries. The optimization currently performed include projection and filter push-down, subquery decorrelation, and other kinds of query rewriting. Old planner does not yet optimize the order of joins, but executes them in the same order as defined in the query (order of Tables in the `FROM` clause and/or order of join predicates in the `WHERE` clause).
+Apache Flink 利用 Apache Calcite 来优化和解析查询。当前执行的优化包括投影和过滤器下推，子查询去相关以及其他类型的查询重写。原版计划程序尚未优化 join 的顺序，而是按照查询中定义的顺序执行它们（FROM 子句中的表顺序和/或 WHERE 子句中的 join 谓词顺序）。
```
Comment (libenchao) [inline docs/dev/table/common.zh.md]: how about “子查询消除”？

File: docs/dev/table/common.zh.md [hunk 8]
```diff
@@ -1406,54 +1385,54 @@ val table: Table = tableEnv.fromDataStream(stream, 'name as 'myName)
...
-* Partition pruning
-* Filter push-down
-* Sub-plan deduplication to avoid duplicate computation
-* Special subquery rewriting, including two parts:
-    * Converts IN and EXISTS into left semi-joins
-    * Converts NOT IN and NOT EXISTS into left anti-join
-* Optional join reordering
-    * Enabled via `table.optimizer.join-reorder-enabled`
+* 基于 Apache Calcite 的子查询解相关
+* 模型剪枝
```
Comment (libenchao) [inline docs/dev/table/common.zh.md]: how about 投影剪裁?

Comment (libenchao) [reply docs/dev/table/common.zh.md]: how about translate pruning  to ”剪裁“ instead of “剪枝" ？

File: docs/dev/table/common.zh.md [hunk 9]
```diff
@@ -1406,54 +1385,54 @@ val table: Table = tableEnv.fromDataStream(stream, 'name as 'myName)
...
 </div>
 
 
-### Explaining a Table
+### 解释表
 
-The Table API provides a mechanism to explain the logical and optimized query plans to compute a `Table`. 
-This is done through the `TableEnvironment.explain(table)` method or `TableEnvironment.explain()` method. `explain(table)` returns the plan of a given `Table`. `explain()` returns the result of a multiple-sinks plan and is mainly used for the Blink planner. It returns a String describing three plans:
+Table API 提供了一种机制来解释计算`表`的逻辑和优化查询计划。
+这是通过 `TableEnvironment.explain(table)` 或者 `TableEnvironment.explain()` 完成的。`explain(table)` 返回给定`表`的计划。 `explain()` 返回多 sink 计划的结果并且主要用于 Blink 计划器。它返回一个描述三中计划的字符串：
```
Comment (libenchao) [inline docs/dev/table/common.zh.md]: 三中计划 -> 三种计划

File: docs/internals/stream_checkpointing.zh.md [hunk 10]
```diff
@@ -23,151 +23,117 @@ specific language governing permissions and limitations
 under the License.
 -->
 
-This document describes Flink's fault tolerance mechanism for streaming data flows.
+本文将介绍 Flink 流数据处理的容错机制（fault tolerance mechanism）。
```
Comment (libenchao) [inline docs/internals/stream_checkpointing.zh.md]: seems that this page is not in this issue， maybe you need to open another pr for this.

**Comment by CrazyTomatoOo** [comment]
@libenchao thanks for your advise, I will revise it as soon as possible.

**Comment by CrazyTomatoOo** [comment]
@wuchong Sure

**Comment by CrazyTomatoOo** [comment]
Hi @libenchao, I have revise the doc, what should I do now?

**Comment by libenchao** [comment]
@CrazyTomatoOo Thanks for the updating. It seems that your pr is conflicted with master, could you rebase your pr towards master to resolve the conflict?

**Comment by CrazyTomatoOo** [comment]
@libenchao，I'm not sure if it's the right way to resolve the conflict. It's my first time. Please check it out. Thanks！

File: docs/dev/table/common.zh.md [hunk 11]
```diff
@@ -1232,9 +1208,9 @@ val table: Table = tableEnv.fromDataStream(stream, 'myLong)
 </div>
 </div>
 
-#### Tuples (Scala and Java) and Case Classes (Scala only)
+#### Tuple类型（Scala 和 Java）和 Case Class类型（仅 Scala）
 
-Flink supports Scala's built-in tuples and provides its own tuple classes for Java. DataStreams and DataSets of both kinds of tuples can be converted into tables. Fields can be renamed by providing names for all fields (mapping based on position). If no field names are specified, the default field names are used. If the original field names (`f0`, `f1`, ... for Flink Tuples and `_1`, `_2`, ... for Scala Tuples) are referenced, the API assumes that the mapping is name-based instead of position-based. Name-based mapping allows for reordering fields and projection with alias (`as`).
+Flink 支持 Scala 的内置 tuple 类型并给 Java 提供自己的 tuple 类型。两种 tuple 的DataStream和DataSet都能被转换成表。可以通过提供所有字段名称来重命名字段（基于位置映射）。如果没有指明任何字段名称，则会使用默认的字段名称。如果引用了原始字段名称（对于 Flink tuple 为`f0`、`f1` ... ...，对于 Scala tuple 为`_1`、`_2` ... ...），则 API 会假定映射是基于名称的而不是基于位置的。基于名称的映射可以通过 `as` 对字段和投影进行重新排序。
```
Comment (libenchao) [inline docs/dev/table/common.zh.md]: add space around `DataStream` `DataSet`

**Comment by CrazyTomatoOo** [comment]
@libenchao, finished.

**Comment by libenchao** [comment]
LGTM. +1 for merging. @CrazyTomatoOo let's wait for a committer to have a look.

**Comment by wuchong** [comment]
I will review and merge it later.

**Pull request closed without merging**